webrtc.html

```diff
@@ -3835,18 +3835,49 @@
               <dd>
-                The <dfn><code>RTCIceCandidate()</code></dfn> constructor takes
+                <p>The <dfn><code>RTCIceCandidate()</code></dfn> constructor takes
                 a dictionary argument, <var>candidateInitDict</var>, whose
                 content is used to initialize the new <a class=
-                "idlType"><code>RTCIceCandidate</code></a> object. When run, if
-                both the <code><a>sdpMid</a></code> and
-                <code><a>sdpMLineIndex</a></code> dictionary members are
-                <code>null</code>, <a>throw</a> a <code>TypeError</code>.
+                "idlType"><code>RTCIceCandidate</code></a> object.</p>
+                <p>When invoked, run the following steps:</p>
+                <ol>
+                  <li>If both the <code><a>sdpMid</a></code> and <code>
+                  <a>sdpMLineIndex</a></code> dictionary members in
+                  <var>candidateInitDict</var> are <code>null</code>, <a>throw</a>
+                  a <code>TypeError</code>.</li>
+                  <li>Let <var>iceCandidate</var> be a newly created
+                  <code>RTCIceCandidate</code> object.</li>
+                  <li>Set the corresponding attributes in <var>iceCandidate</var>
+                  with the dictionary member values of <var>candidateInitDict</var></li>
+                  <li>Let <var>candidate</var> be the <code><a>candidate</a></code>
+                  dictionary member of <var>candidateInitDict</var>. If
+                  <var>candidate</var> is not an empty string, run the following steps:
+                  <ol>
+                    <li>Parse <var>candidate</var> using the grammar for
+                    <code>candidate-attribute</code> as defined in section 15.1 of
+                    [[!ICE]], together with the grammar extension for ICE TCP as
+                    defined in section 4.5 of [[!RFC6544]].</li>
+                    <li>If parsing of <code>candidate-attribute</code> is successful,
+                    set the corresponding attributes in <var>iceCandidate</var> to
+                    the field values of the parse result.</li>
+                  </ol></li>
+                  <li>Return <var>iceCandidate</var></li>
+                </ol>
+                <div class="note">
+                  The validation for members in <var>candidateInitDict</var>
+                  is not done in the constructor for <code>RTCIceCandidate</code>.
+                  Instead, validation is done when calling <code><a data-for=
+                  "RTCPeerConnection">addIceCandidate()</a></code>. To maintain
+                  backward compatibility, if the <var>candidate</var> member do not
+                  match the grammar for <code>candidate-attribute</code>, the value
+                  remain set in the <code>candidate</code> attribute with the other
+                  derivative fields set to <code>null</code>
```

Comment:
Meaning, the DOMString `candidate` attribute will return the string passed in, but the parsed attributes will return `null`? I think this could be more clear.